sensors/gps/pa6h.c

```diff
@@ -176,19 +450,89 @@ static int pa6h_alloc(sensor_info_t *info, const char *args)
 		return err;
 	}
 
-	ctx->filedes = open(path, O_RDONLY | O_NOCTTY | O_NONBLOCK);
-	while (ctx->filedes < 0) {
-		usleep(10 * 1000);
-		cnt++;
+	/* Opening serial device */
+	do {
+		ctx->filedes = open(path, O_RDWR | O_NOCTTY | O_NONBLOCK);
 
-		if (cnt > 10000) {
-			fprintf(stderr, "pa6h: Can't open %s: %s\n", path, strerror(errno));
-			err = -errno;
-			free(ctx);
-			return err;
+		if (ctx->filedes < 0) {
+			usleep(10 * 1000);
+			cnt++;
+
+			if (cnt > 10000) {
+				fprintf(stderr, "%s Can't open %s: %s\n", PA6H_STR, path, strerror(errno));
+				err = -errno;
+				free(ctx);
+				return err;
+			}
 		}
-		ctx->filedes = open(path, O_RDONLY | O_NOCTTY | O_NONBLOCK);
+	} while (ctx->filedes < 0);
+
+	if (isatty(ctx->filedes) != 1) {
+		fprintf(stderr, "%s %s not a tty\n", PA6H_STR, path);
+		close(ctx->filedes);
+		return -1;
 	}
+
+	/* First serial setup backups termios settings */
+	if (pa6h_serialSetup(ctx->filedes, B9600, &termBackup) < 0) {
+		printf("%s cannot set baud %d\n", PA6H_STR, 9600);
+		close(ctx->filedes);
+		return -1;
+	}
+
+	/* Bruteforce baudrate 115200. GPS changes baud instantly and we certainly will miss output message (no output check) */
+	cnt = 0;
+	while (cnt++ < repeat) {
+		pa6h_paramSet(ctx->filedes, PMTK_ID_SET_NMEA_BAUDRATE, PMTK_SET_NMEA_BAUDRATE_115200, 0);
```

Comment:
Initially i thought that baudrate setting message is acknowledged by gps receiver but we are unable to catch it. However I have found in one documentation that it is not. This allows for function refactor so that `ack` is returned via an argument, not a return value. Return value signals OS related error in transmission.

Now return value is checked.